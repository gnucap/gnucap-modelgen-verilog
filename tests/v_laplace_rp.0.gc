
attach ../vams/vsine.so

verilog

va_laplace_rp #(.rr0(2)) b0(0,out1,in,0);
// va_laplace #(.pr0(1.)) b0(0,out2,in,0); // incomplete
va_laplace_rp #(.rr0(1.), .pr0(1.)) b0(0,out3,in,0);
va_laplace_rp #(.rr0(1.), .pr0(1.), .rr1(1.), .pr1(2.)) b0(0,out4,in,0);
resistor #(1.) r1(out1, 0);
resistor #(1.) r2(out2, 0);
resistor #(1.) r3(out3, 0);
resistor #(1.) r4(out4, 0);

list
dc
list

vsine #(.ampl(1), .freq(1)) v1(in, 0);
print tran v(nodes)
tran 1 basic

print ac vm(nodes) vp(nodes)
ac .01 100 * 10

status notime
