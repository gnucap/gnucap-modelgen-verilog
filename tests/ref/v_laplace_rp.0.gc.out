va_laplace #(.rr0(2)) b0 (.out0(0),.out1(out1),.in0(in),.in1(0));
va_laplace_nd #(.n0( 2.),.d0( 0.),.d1( 1.)) b0 (.out0(0),.out1(ref1),.in0(in),.in1(0));
va_laplace #(.pr0(1.)) b0 (.out0(0),.out1(out2),.in0(in),.in1(0));
va_laplace #(.rr0(1.),.pr0(1.)) b0 (.out0(0),.out1(out3),.in0(in),.in1(0));
va_laplace_nd #(.n0( 1.),.d0( -1.),.d1( 1.)) b0 (.out0(0),.out1(ref3),.in0(in),.in1(0));
va_laplace #(.rr0(1.),.rr1(1.),.pr0(1.),.pr1(2.)) b0 (.out0(0),.out1(out4),.in0(in),.in1(0));
va_laplace_nd #(.n0(-3.),.n1( 2.),.d0( 2.),.d1(-3.),.d2( 1.)) b0 (.out0(0),.out1(ref4),.in0(in),.in1(0));
resistor #(.r(1.)) r1 (.p(out1),.n(0));
resistor #(.r(1.)) r2 (.p(out2),.n(0));
resistor #(.r(1.)) r3 (.p(out3),.n(0));
resistor #(.r(1.)) r4 (.p(out4),.n(0));
resistor #(.r(1.)) r1 (.p(ref1),.n(0));
resistor #(.r(1.)) r2 (.p(ref2),.n(0));
resistor #(.r(1.)) r3 (.p(ref3),.n(0));
resistor #(.r(1.)) r4 (.p(ref4),.n(0));
#          
 0.        
va_laplace #(.n0( 2.),.d0( 0.),.d1( 1.)) b0 (.out0(0),.out1(out1),.in0(in),.in1(0));
va_laplace_nd #(.n0( 2.),.d0( 0.),.d1( 1.)) b0 (.out0(0),.out1(ref1),.in0(in),.in1(0));
va_laplace #(.n0( 0.),.d0(-1.),.d1( 1.)) b0 (.out0(0),.out1(out2),.in0(in),.in1(0));
va_laplace #(.n0( 1.),.d0(-1.),.d1( 1.)) b0 (.out0(0),.out1(out3),.in0(in),.in1(0));
va_laplace_nd #(.n0( 1.),.d0( -1.),.d1( 1.)) b0 (.out0(0),.out1(ref3),.in0(in),.in1(0));
va_laplace #(.n0(-3.),.n1( 2.),.d0( 2.),.d1(-3.),.d2( 1.)) b0 (.out0(0),.out1(out4),.in0(in),.in1(0));
va_laplace_nd #(.n0(-3.),.n1( 2.),.d0( 2.),.d1(-3.),.d2( 1.)) b0 (.out0(0),.out1(ref4),.in0(in),.in1(0));
resistor #(.r(1.)) r1 (.p(out1),.n(0));
resistor #(.r(1.)) r2 (.p(out2),.n(0));
resistor #(.r(1.)) r3 (.p(out3),.n(0));
resistor #(.r(1.)) r4 (.p(out4),.n(0));
resistor #(.r(1.)) r1 (.p(ref1),.n(0));
resistor #(.r(1.)) r2 (.p(ref2),.n(0));
resistor #(.r(1.)) r3 (.p(ref3),.n(0));
resistor #(.r(1.)) r4 (.p(ref4),.n(0));
#Time       v(in)      v(out1)    v(out2)    v(out3)    v(out4)    v(ref1)    v(ref2)    v(ref3)    v(ref4)   
 0.         0.         0.         0.         0.         0.         0.         0.         0.         0.        
 89.508E-9  562.39E-9  101.E-15   0.         50.E-15    101.E-15   101.E-15   0.         50.E-15    101.E-15  
 318.8E-9   2.0031E-6  689.E-15   0.         344.E-15   689.E-15   689.E-15   0.         344.E-15   689.E-15  
 548.09E-9  3.4438E-6  1.938E-12  0.         969.E-15   1.938E-12  1.938E-12  0.         969.E-15   1.938E-12 
 777.38E-9  4.8845E-6  3.847E-12  0.         1.924E-12  3.847E-12  3.847E-12  0.         1.924E-12  3.847E-12 
 25.359E-6  159.34E-6  4.0408E-9  0.         2.0204E-9  4.0409E-9  4.0408E-9  0.         2.0204E-9  4.0409E-9 
 49.942E-6  313.79E-6  15.671E-9  0.         7.8358E-9  15.672E-9  15.671E-9  0.         7.8358E-9  15.672E-9 
 88.902E-6  558.59E-6  49.659E-9  0.         24.83E-9   49.662E-9  49.659E-9  0.         24.83E-9   49.662E-9 
 127.86E-6  803.38E-6  102.72E-9  0.         51.363E-9  102.73E-9  102.72E-9  0.         51.363E-9  102.73E-9 
 166.82E-6  0.0010482  174.86E-9  0.         87.434E-9  174.87E-9  174.86E-9  0.         87.434E-9  174.87E-9 
 205.78E-6  0.001293   266.07E-9  0.         133.04E-9  266.1E-9   266.07E-9  0.         133.04E-9  266.1E-9  
 244.74E-6  0.0015378  376.35E-9  0.         188.19E-9  376.4E-9   376.35E-9  0.         188.19E-9  376.4E-9  
 303.94E-6  0.0019097  580.43E-9  0.         290.24E-9  580.52E-9  580.43E-9  0.         290.24E-9  580.52E-9 
 363.13E-6  0.0022816  828.54E-9  0.         414.32E-9  828.69E-9  828.54E-9  0.         414.32E-9  828.69E-9 
 452.82E-6  0.0028451  1.2883E-6  0.         644.27E-9  1.2886E-6  1.2883E-6  0.         644.27E-9  1.2886E-6 
 542.5E-6   0.0034087  1.8492E-6  0.         924.77E-9  1.8497E-6  1.8492E-6  0.         924.77E-9  1.8497E-6 
 632.19E-6  0.0039722  2.5112E-6  0.         1.2558E-6  2.512E-6   2.5112E-6  0.         1.2558E-6  2.512E-6  
 784.79E-6  0.004931   3.8698E-6  0.         1.9354E-6  3.8713E-6  3.8698E-6  0.         1.9354E-6  3.8713E-6 
 937.39E-6  0.0058897  5.521E-6   0.         2.7614E-6  5.5236E-6  5.521E-6   0.         2.7614E-6  5.5236E-6 
 0.00117    0.0073512  8.601E-6   0.         4.3022E-6  8.6061E-6  8.601E-6   0.         4.3022E-6  8.6061E-6 
 0.0014026  0.0088127  12.361E-6  0.         6.1834E-6  12.37E-6   12.361E-6  0.         6.1834E-6  12.37E-6  
 0.0016352  0.010274   16.801E-6  0.         8.405E-6   16.815E-6  16.801E-6  0.         8.405E-6   16.815E-6 
 0.0020297  0.012752   25.884E-6  0.         12.951E-6  25.91E-6   25.884E-6  0.         12.951E-6  25.91E-6  
 0.0024241  0.015231   36.922E-6  0.         18.476E-6  36.967E-6  36.922E-6  0.         18.476E-6  36.967E-6 
 0.0030251  0.019006   57.496E-6  0.         28.777E-6  57.584E-6  57.496E-6  0.         28.777E-6  57.584E-6 
 0.003626   0.022781   82.608E-6  0.         41.354E-6  82.759E-6  82.608E-6  0.         41.354E-6  82.759E-6 
 0.004227   0.026556   112.26E-6  0.         56.208E-6  112.5E-6   112.26E-6  0.         56.208E-6  112.5E-6  
 0.0052451  0.03295    172.84E-6  0.         86.575E-6  173.3E-6   172.84E-6  0.         86.575E-6  173.3E-6  
 0.0062633  0.039343   246.45E-6  0.         123.49E-6  247.23E-6  246.45E-6  0.         123.49E-6  247.23E-6 
 0.0078112  0.04906    383.29E-6  0.         192.15E-6  384.8E-6   383.29E-6  0.         192.15E-6  384.8E-6  
 0.0093591  0.058771   550.2E-6   0.         275.96E-6  552.8E-6   550.2E-6   0.         275.96E-6  552.8E-6  
 0.010907   0.068477   747.16E-6  0.         374.95E-6  751.28E-6  747.16E-6  0.         374.95E-6  751.28E-6 
 0.013524   0.084869   0.0011484  0.         576.82E-6  0.0011563  0.0011484  0.         576.82E-6  0.0011563 
 0.01614    0.10124    0.0016354  0.         822.14E-6  0.0016488  0.0016354  0.         822.14E-6  0.0016488 
 0.020107   0.126      0.0025369  0.         0.0012771  0.0025628  0.0025369  0.         0.0012771  0.0025628 
 0.024075   0.15069    0.0036346  0.         0.0018321  0.0036791  0.0036346  0.         0.0018321  0.0036791 
 0.028042   0.17528    0.0049278  0.         0.0024872  0.0049981  0.0049278  0.         0.0024872  0.0049981 
 0.034699   0.2163     0.0075346  0.         0.0038116  0.0076683  0.0075346  0.         0.0038116  0.0076683 
 0.041356   0.25693    0.010685   0.         0.0054175  0.010912   0.010685   0.         0.0054175  0.010912  
 0.051447   0.31765    0.016483   0.         0.0083863  0.016921   0.016483   0.         0.0083863  0.016921  
 0.061538   0.37709    0.023494   0.         0.011994   0.024244   0.023494   0.         0.011994   0.024244  
 0.071629   0.43502    0.031689   0.         0.016234   0.032873   0.031689   0.         0.016234   0.032873  
 0.088509   0.52789    0.047942   0.         0.024707   0.050183   0.047942   0.         0.024707   0.050183  
 0.10539    0.61483    0.067231   0.         0.034854   0.071015   0.067231   0.         0.034854   0.071015  
 0.13095    0.73304    0.10168    0.         0.053205   0.10894    0.10168    0.         0.053205   0.10894   
 0.15651    0.83237    0.1417     0.         0.074848   0.15404    0.1417     0.         0.074848   0.15404   
 0.19485    0.94056    0.20967    0.         0.11243    0.23327    0.20967    0.         0.11243    0.23327   
 0.23319    0.99443    0.28386    0.         0.15464    0.3237     0.28386    0.         0.15464    0.3237    
 0.29218    0.96509    0.39944    0.         0.22359    0.47525    0.39944    0.         0.22359    0.47525   
 0.35116    0.80471    0.50383    0.         0.29096    0.62964    0.50383    0.         0.29096    0.62964   
 0.41015    0.53505    0.58286    0.         0.34935    0.77249    0.58286    0.         0.34935    0.77249   
 0.49441    0.03511    0.63091    0.         0.40517    0.93241    0.63091    0.         0.40517    0.93241   
 0.57868   -0.47445    0.59388    0.         0.42149    1.0255     0.59388    0.         0.42149    1.0255    
 0.68401   -0.91526    0.44751    0.         0.39109    1.0556     0.44751    0.         0.39109    1.0556    
 0.78934   -0.96961    0.24897    0.         0.32979    1.0398     0.24897    0.         0.32979    1.0398    
 0.89467   -0.61455    0.082109   0.         0.27839    1.0623     0.082109   0.         0.27839    1.0623    
 0.94733   -0.3249     0.032633   0.         0.26804    1.113      0.032633   0.         0.26804    1.113     
 1.         14.E-15    0.015522   0.         0.27375    1.2036     0.015522   0.         0.27375    1.2036    
#Freq       vm(in)     vm(out1)   vm(out2)   vm(out3)   vm(out4)   vm(ref1)   vm(ref2)   vm(ref3)   vm(ref4)   vp(in)     vp(out1)   vp(out2)   vp(out3)   vp(out4)   vp(ref1)   vp(ref2)   vp(ref3)   vp(ref4)  
 0.01       1.         31.831     0.         0.99803    1.4976     31.831     0.         0.99803    1.4976     0.        -90.        0.        -176.4     -177.      -90.        0.        -176.4     -177.      
 0.1        1.         3.1831     0.         0.84673    1.3137     3.1831     0.         0.84673    1.3137     0.        -90.        0.        -147.86    -153.15    -90.        0.        -147.86    -153.15    
 1.         1.         0.31831    0.         0.15718    0.30796    0.31831    0.         0.15718    0.30796    0.        -90.        0.        -99.043    -103.27    -90.        0.        -99.043    -103.27    
 10.        1.         0.031831   0.         0.015913   0.03182    0.031831   0.         0.015913   0.03182    0.        -90.        0.        -90.912    -91.367    -90.        0.        -90.912    -91.367    
 100.       1.         0.0031831  0.         0.0015915  0.0031831  0.0031831  0.         0.0015915  0.0031831  0.        -90.        0.        -90.091    -90.137    -90.        0.        -90.091    -90.137    
Gnucap   System status
iterations: op=0, dc=3, tran=196, fourier=0, total=268
transient timesteps: accepted=58, rejected=6, total=64
nodes: user=12, subckt=0, model=17, total=29
dctran density=36.7%, ac density=10.1%
